modules/local_storage/src/LocalStockServiceConfig.php

```diff
@@ -34,17 +35,17 @@ public function __construct(EntityTypeManagerInterface $entity_type_manager) {
   /**
    * {@inheritdoc}
    *
-   * @ToDo Do we really need a primary transaction location. If so users need a way to set this.
    */
-  public function getPrimaryTransactionLocation(PurchasableEntityInterface $entity, $quantity) {
-    $locations = $this->getLocationList($entity);
+  public function getTransactionLocation(Context $context, PurchasableEntityInterface $entity, $quantity) {
+    // location.
+    $locations = $this->getAvailabilityLocations($context, $entity);
     return empty($locations) ? NULL : array_shift($locations);
```

Comment:
The idea was to allow for different configuration modules with an out of the box one that provides a default/primary location (now by store).  
Getting the "first location with availability" would be a great options but should not be the default out of the box one. I think most sites are ok with checking multiple locations for stock but want the transactions created against a specific one.